src/Kafka/Internal/Setup.hs

```diff
@@ -55,7 +61,7 @@ newTopicConf :: IO TopicConf
 newTopicConf = TopicConf <$> newRdKafkaTopicConfT
 
 newKafkaConf :: IO KafkaConf
-newKafkaConf = KafkaConf <$> newRdKafkaConfT <*> newCancellationToken
+newKafkaConf = KafkaConf <$> newRdKafkaConfT <*> newIORef Nothing <*> newCancellationToken
```

Comment:
This `IORef` is visible for the callbacks, but it is still empty until the queue is created later.